File: ziteng/Minimum Subarray.java
public class Solution {
    /**
     * @param nums: a list of integers
     * @return: A integer indicate the sum of minimum subarray
     */
    public int minSubArray(ArrayList<Integer> nums) {
        int newsum=nums.get(0);
        int min=nums.get(0);
        for(int i=1;i<nums.size();i++){
           newsum=Math.min(newsum+nums.get(i),nums.get(i));
           min= Math.min(min, newsum);
        }
        return min;
    }
}

File: mu_wang/9_10/Minimum Subarray.py
class Solution:
    """
    @param nums: a list of integers
    @return: A integer denote the sum of minimum subarray
    """
    def minSubArray(self, nums):
        # write your code here
        if len(nums)<1:
            return 0
        curMin = nums[0]
        result = nums[0]
        for i in range(1,len(nums)):
            curMin = min(nums[i],curMin+nums[i])
            result = min(result,curMin)
        
        return result